src/sdk/main/include/impl/MirrorNodeContractQuery.h

```diff
@@ -0,0 +1,264 @@
+// SPDX-License-Identifier: Apache-2.0
+#ifndef HIERO_SDK_CPP_MIRROR_NODE_CONTRACT_QUERY_H_
+#define HIERO_SDK_CPP_MIRROR_NODE_CONTRACT_QUERY_H_
+
+#include "AccountId.h"
+#include "Client.h"
+#include "ContractFunctionParameters.h"
+#include "ContractId.h"
+#include "impl/HexConverter.h"
+
+#include <cstdint>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include <nlohmann/json.hpp>
+
+using json = nlohmann::json;
+
+namespace Hiero
+{
+/**
+ * MirrorNodeContractQuery returns a result from EVM execution such as cost-free execution of read-only smart
+ * contract queries, gas estimation, and transient simulation of read-write operations.
+ */
+class MirrorNodeContractQuery
+{
+public:
+  /**
+   * Default constructor.
+   */
+  MirrorNodeContractQuery() = default;
+
+  virtual ~MirrorNodeContractQuery() = default;
+
+  /**
+   * Gets the contract ID of the contract for the Mirror Node call.
+   *
+   * @return The contract ID as an optional string.
+   */
+  const std::optional<ContractId>& getContractId() const { return mContractId; }
+
+  /**
+   * Sets the contract ID of the contract for the Mirror Node call.
+   *
+   * @param id The contract ID.
+   * @return Reference to the updated object.
+   */
+  MirrorNodeContractQuery& setContractId(const ContractId& id);
+
+  /**
+   * Gets the EVM address of the contract for the Mirror Node call.
+   *
+   * @return The contract EVM address as an optional string.
+   */
+  const std::optional<std::string>& getContractEvmAddress() const { return mContractEvmAddress; }
+
+  /**
+   * Sets the EVM address of the contract for the Mirror Node call.
+   *
+   * @param address The contract EVM address.
+   * @return Reference to the updated object.
+   */
+  MirrorNodeContractQuery& setContractEvmAddress(const std::string& address);
+
+  /**
+   * Gets the sender account ID for the Mirror Node call.
+   *
+   * @return The sender account ID as an optional string.
+   */
+  const std::optional<AccountId>& getSender() const { return mSender; }
+
+  /**
+   * Sets the sender account ID for the Mirror Node call.
+   *
+   * @param id The sender account ID.
+   * @return Reference to the updated object.
+   */
+  MirrorNodeContractQuery& setSender(const AccountId& id);
+
+  /**
+   * Gets the sender's EVM address for the Mirror Node call.
+   *
+   * @return The sender's EVM address as an optional string.
+   */
+  const std::optional<std::string>& getSenderEvmAddress() const { return mSenderEvmAddress; }
+
+  /**
+   * Sets the sender's EVM address for the Mirror Node call.
+   *
+   * @param address The sender's EVM address.
+   * @return Reference to the updated object.
+   */
+  MirrorNodeContractQuery& setSenderEvmAddress(const std::string& address);
+
+  /**
+   * Gets the call data for the Mirror Node call.
+   *
+   * @return The call data as a vector of bytes.
+   */
+  const std::vector<std::byte>& getCallData() const { return mCallData; }
```

Comment:
I wouldn't return `const` references to internal members. This unnecessarily limits what the user can do with that data.